NitroxModel/Discovery/GameInstallationFinder.cs

```diff
@@ -1,59 +1,71 @@
-        public static GameInstallationFinder Instance => instance.Value;
-
-        /// <summary>
-        ///     The order of these finders is VERY important. Only change if you know what you're doing.
-        /// </summary>
-        private readonly IFindGameInstallation[] finders = {
-            new GameInCurrentDirectoryFinder(),
-            new ConfigGameFinder(),
-            new SteamGameRegistryFinder(),
-            new EpicGamesInstallationFinder(),
-            new DiscordGameFinder(),
-            new EnvironmentGameFinder()
-        };
-
-        public string FindGame(IList<string> errors = null)
-        {
-            errors ??= new List<string>();
-            foreach (IFindGameInstallation finder in finders)
-            {
-                string path = finder.FindGame(errors);
-                if (path == null)
-                {
-                    continue;
-                }
-
-                errors.Clear();
-                return Path.GetFullPath(path);
-            }
+        { GameLibraries.CURRENT_DIRECTORY, new GameInCurrentDirectoryFinder() },
+        { GameLibraries.CONFIG, new ConfigGameFinder() },
+        { GameLibraries.STEAM, new SteamGameRegistryFinder() },
+        { GameLibraries.EPIC, new EpicGamesInstallationFinder() },
+        { GameLibraries.DISCORD, new DiscordGameFinder() },
+        { GameLibraries.MICROSOFT, null },
+        { GameLibraries.ENVIRONMENT, new EnvironmentGameFinder() },
+    };
+
+    public IEnumerable<GameInstallation> FindGame(GameInfo gameInfo, GameLibraries gameLibraries, IList<string> errors)
+    { 
+        errors ??= new List<string>();
 
-            return null;
+        if (gameInfo == null || gameLibraries == GameLibraries.UNKNOWN)
+        {
+            yield break;
         }
 
-        public static bool IsSubnauticaDirectory(string directory)
+        foreach (GameLibraries wantedFinder in GetFlags(gameLibraries))
         {
-            if (string.IsNullOrWhiteSpace(directory))
+            if (!finders.TryGetValue(wantedFinder, out IGameFinder finder))
             {
-                return false;
+                errors.Add($"Could not find game finder for configuration : '{wantedFinder}'");
+                continue;
             }
 
-            return Directory.EnumerateFiles(directory, "*.exe")
-                .Any(file => Path.GetFileName(file)?.Equals("subnautica.exe", StringComparison.OrdinalIgnoreCase) ?? false);
+            GameInstallation? finderResult = finder.FindGame(gameInfo, errors);
+            if (finderResult.HasValue)
+            {
+                yield return finderResult.Value;
+            }
         }
     }
+
+    public static T[] GetFlags<T>(T flagsEnumValue) where T : Enum
+    {
+        return Enum
+            .GetValues(typeof(T))
+            .Cast<T>()
+            .Where(e => flagsEnumValue.HasFlag(e))
+            .ToArray();
+    }
+
+    public static bool IsSubnauticaDirectory(string path, GameInfo gameInfo)
```

Comment:
Should be inlined as it's one line that does not do exactly what the method name says.